share/namespace.go

```diff
@@ -124,7 +130,10 @@ func (n Namespace) ValidateForBlob() error {
 	if err := n.ValidateForData(); err != nil {
 		return err
 	}
-	if bytes.Compare(n, MaxReservedNamespace) < 1 {
+	if bytes.Compare(n, MaxPrimaryReservedNamespace) < 1 {
+		return fmt.Errorf("invalid blob namespace(%s): reserved namespaces are forbidden", n)
+	}
+	if bytes.Compare(n, MinSecondaryReservedNamespace) > -1 {
```

Comment:
[optional][no change needed] this could also be https://github.com/celestiaorg/celestia-node/pull/2562/files#diff-46c85ee9c611a9b9c56eb104a6dca5bcbc6b3961683cdf0c941704b18496084dR129 so that if celestia-app makes changes to namespace validation, celestia-node gets those changes.